pkg/karpenter/providers/helm/helm_test.go

```diff
@@ -0,0 +1,265 @@
+			buffer, err := dummyIndexFile()
+			Expect(err).NotTo(HaveOccurred())
+			fakeGetter.GetReturns(buffer, nil)
+			installer := Installer{
+				Getters: getters,
+				Settings: &cli.EnvSettings{
+					RegistryConfig:   filepath.Join(tmp, "registry.json"),
+					RepositoryConfig: filepath.Join(tmp, "repositories.yaml"),
+					RepositoryCache:  tmp,
+				},
+			}
+			err = installer.AddRepo("https://charts.karpenter.sh", "karpenter")
+			Expect(err).NotTo(HaveOccurred())
+			content, err := os.ReadFile(filepath.Join(tmp, "repositories.yaml"))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(content)).To(Equal(expectedRepositoryYaml))
+		})
+		When("the getter fails to retrieve the index file", func() {
+			It("returns an error", func() {
+				fakeGetter.GetReturns(nil, errors.New("nope"))
+				installer := Installer{
+					Getters: getters,
+					Settings: &cli.EnvSettings{
+						RegistryConfig:   filepath.Join(tmp, "registry.json"),
+						RepositoryConfig: filepath.Join(tmp, "repositories.yaml"),
+						RepositoryCache:  tmp,
+					},
+				}
+				err = installer.AddRepo("https://charts.karpenter.sh", "karpenter")
+				Expect(err).To(MatchError(ContainSubstring("failed to download index file: nope")))
+			})
+		})
+		When("the getter returns an invalid JSON", func() {
+			It("returns an error", func() {
+				buffer := bytes.NewBuffer([]byte("invalid"))
+				fakeGetter.GetReturns(buffer, nil)
+				installer := Installer{
+					Getters: getters,
+					Settings: &cli.EnvSettings{
+						RegistryConfig:   filepath.Join(tmp, "registry.json"),
+						RepositoryConfig: filepath.Join(tmp, "repositories.yaml"),
+						RepositoryCache:  tmp,
+					},
+				}
+				err = installer.AddRepo("https://charts.karpenter.sh", "karpenter")
+				Expect(err).To(MatchError(ContainSubstring("failed to download index file: error unmarshaling JSON")))
+			})
+		})
+		When("the repository url is invalid", func() {
+			It("returns an error", func() {
+				installer := Installer{
+					Getters: getters,
+					Settings: &cli.EnvSettings{
+						RegistryConfig:   filepath.Join(tmp, "registry.json"),
+						RepositoryConfig: filepath.Join(tmp, "repositories.yaml"),
+						RepositoryCache:  tmp,
+					},
+				}
+				err = installer.AddRepo("%^&", "karpenter")
+				Expect(err).To(MatchError(ContainSubstring("invalid chart URL format: %^&")))
+			})
+		})
+		When("there is no provider for the given scheme", func() {
+			It("returns an error", func() {
+				installer := Installer{
+					Getters: nil,
+					Settings: &cli.EnvSettings{
+						RegistryConfig:   filepath.Join(tmp, "registry.json"),
+						RepositoryConfig: filepath.Join(tmp, "repositories.yaml"),
+						RepositoryCache:  tmp,
+					},
+				}
+				err = installer.AddRepo("https://charts.karpenter.sh", "karpenter")
+				Expect(err).To(MatchError(ContainSubstring("failed to create new chart repository: could not find protocol handler for: ")))
+			})
+		})
+	})
+
+	Context("InstallChart", func() {
```

Comment:
missing an error test scenario? I see 3 `return err` scenarios but only 2 tests :smile:

Reply:
Ah! I wanted to extend this. It's in my notes. :D After I found out how the f*ck to mock Kubernetes. Luckily, Helm did it for us ah.